gradle/plugins/src/main/kotlin/com.hedera.gradle.java.gradle.kts

```diff
@@ -0,0 +1,170 @@
+    }
+
+    // propagate system environment to test runner
+    systemProperty("OPERATOR_ID", providers.gradleProperty("OPERATOR_ID").getOrElse(""))
+    systemProperty("OPERATOR_KEY", providers.gradleProperty("OPERATOR_KEY").getOrElse(""))
+    systemProperty("CONFIG_FILE", providers.gradleProperty("CONFIG_FILE").getOrElse(""))
+    systemProperty("HEDERA_NETWORK", providers.gradleProperty("HEDERA_NETWORK").getOrElse(""))
+}
+
+tasks.withType<Javadoc>().configureEach {
+    options {
+        this as StandardJavadocDocletOptions
+        encoding = "UTF-8"
+        addStringOption("Xdoclint:all,-missing", "-quiet")
+        addStringOption("Xwerror", "-quiet")
+    }
+}
+
+tasks.jacocoTestReport {
+    // make sure to use any/all test coverage data for the report and run all tests before this report is made
+    executionData.from(
+        tasks.test.map { it.extensions.getByType<JacocoTaskExtension>().destinationFile!! },
+        tasks.named("testIntegration").map { it.extensions.getByType<JacocoTaskExtension>().destinationFile!! }
+    )
+
+    // remove generated proto files from report
+    classDirectories.setFrom(sourceSets.main.get().output.asFileTree.matching {
+        exclude(
+            "**/proto/**",
+            "**/AccountAllowanceAdjustTransaction.*",
+            "**/HederaPreCheckStatusException.*",
+            "**/HederaReceiptStatusException.*"
+        )
+    })
+
+    // configure it so only xml is generated for the report
+    reports {
+        xml.required = true
+        html.required = true
+        csv.required = false
+    }
+}
+
+
+// https://github.com/google/protobuf-gradle-plugin
+protobuf {
+    protoc {
+        artifact = "com.google.protobuf:protoc:3.21.9"
+    }
+    plugins {
+        id("grpc") {
+            artifact = "io.grpc:protoc-gen-grpc-java:1.50.2"
+        }
+    }
+}
+tasks.generateProto {
+    plugins { plugins.register("grpc") }
+}
+
+tasks.compileJava {
+    options.javaModuleVersion = project.version.toString()
+}
+
+tasks.jar {
+    exclude("**/*.proto")
+    includeEmptyDirs = false
+    manifest {
+        attributes["Implementation-Version"] = project.version
+    }
+}
+
+sourceSets.all {
+    configurations[getTaskName("", "compileProtoPath")].extendsFrom(configurations["internal"])
+}
+
+spotbugs {
+    //ignoreFailures = false
+    //showStackTraces = true
+    //showProgress = false
+    //reportLevel = 'default'
+    //effort = 'default'
+    //visitors = [ 'FindSqlInjection', 'SwitchFallthrough' ]
+    //omitVisitors = [ 'FindNonShortCircuit' ]
+    reportsDir = layout.buildDirectory.dir("reports/spotbugs")
+    //includeFilter = file('spotbugs-include.xml')
+    //excludeFilter = file('spotbugs-exclude.xml')
+    onlyAnalyze = listOf("com.hedera.hashgraph.sdk.*")
+    //projectName = name
+    //release = version
+    //extraArgs = [ '-nested:false' ]
+    //jvmArgs = [ '-Duser.language=ja' ]
+    //maxHeapSize = '512m'
+}
```

Comment:
Why are we leaving all these commented out things in here? Just shortcuts for various debugging scenarios?

Reply:
@rbarkerSL this is legacy things as of now. Most probably we will remove spotbugs completely, but we will review it as a follow up, because this thing is huge already:)